prov/efa/src/efa_fabric.c

```diff
@@ -1034,10 +1034,11 @@ static int efa_init_info(const struct fi_info **all_infos)
 	for (i = 0; i < num_devices; i++) {
 		ret = efa_alloc_info(ctx_list[i], &fi, &efa_rdm_domain);
 		if (!ret) {
-			if (!*all_infos)
+			if (!*all_infos) {
 				*all_infos = fi;
-			else
+			} else if (tail) {
```

Comment:
Not sure the commit message match the change here. The change is to check `tail` right?

Reply:
I think assertion is more appropriate here.

Reply:
+1